src/app/operators/components/operator/operator.component.ts

```diff
@@ -2,28 +2,56 @@ import {
   Component,
   Input,
   OnInit,
-  ChangeDetectionStrategy
-} from "@angular/core";
-import { OperatorDoc } from "../../../../operator-docs/operator.model";
+  ChangeDetectionStrategy,
+  Inject,
+  InjectionToken
+} from '@angular/core';
+import { Router, ActivatedRoute } from '@angular/router';
+import { SeoService } from '../../../services/seo.service';
+import { OperatorDoc } from '../../../../operator-docs/operator.model';
+import 'rxjs/add/operator/pluck';
+
+export const OPERATOR_TOKEN = new InjectionToken<string>('operators');
 
 @Component({
-  selector: "app-operator",
-  templateUrl: "./operator.component.html",
-  styleUrls: ["./operator.component.scss"],
-  changeDetection: ChangeDetectionStrategy.OnPush
+  selector: 'app-operator',
+  templateUrl: './operator.component.html',
+  styleUrls: ['./operator.component.scss']
 })
-export class OperatorComponent {
-  @Input() operator: OperatorDoc;
-
-  private readonly baseSourceUrl = "https://github.com/ReactiveX/rxjs/blob/master/src/operators/";
-  private readonly baseSpecUrl = "http://reactivex.io/rxjs/test-file/spec-js/operators";
+export class OperatorComponent implements OnInit {
+  public operator: OperatorDoc;
+
+  private readonly baseSourceUrl = 'https://github.com/ReactiveX/rxjs/blob/master/src/operators/';
+  private readonly baseSpecUrl = 'http://reactivex.io/rxjs/test-file/spec-js/operators';
+
+  constructor(
+    private _router: Router,
+    private _activatedRoute: ActivatedRoute,
+    @Inject(OPERATOR_TOKEN) public operators: OperatorDoc[],
+    private _seo: SeoService
```

Comment:
Since this is an Angular project I believe it's worth sticking to their official style guide that discourages from prefixing private variables with `_`. See https://angular.io/guide/styleguide#style-03-04

Reply:
This is debatable, in the material docs project `_` is being used for private. I prefer this convention. https://github.com/angular/material.angular.io/blob/a742dc087b1e8327fe3f0a6effa558e3e301dabe/src/app/shared/doc-viewer/doc-viewer.ts#L24-L25

Reply:
I prefer to stick to the existing code in the project and use _